gateway/src/apicast/proxy.lua

```diff
@@ -355,6 +355,14 @@ local function limit_reset(response_headers)
   return response_headers and response_headers['3scale-limit-reset']
 end
 
+-- Returns the '3scale-limit-max-value' from the headers of a 3scale backend
+-- response.
+-- This header is set only when enabled via the '3scale-options' header of the
+-- request.
+local function limit_max_value(response_headers)
+  return response_headers and response_headers['3scale-limit-max-value']
```

Comment:
I would change here to return tonumber(response_headers['3scale-limit-max-value'])

Reply:
It would still need to be `tonumber(response_headers  and response_headers['3scale-limit-max-value'])`.